Reorganised static templates and logic

MebinAbraham opened the pull request:
### What is the context of this PR?
- Added 404 and 500 error handlers using the factory pattern
- Added 404 and 500 error page HTML templates
- Renders survey long name for more context
- Fixed some broken logic with Ajax due to previous refactors

### How to review 
Make sure app works and behaves as expected.

AdamHawtin commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Create error page for front end](https://trello.com/c/fCmMpyBd/28-create-error-page-for-front-end)

Commit:
Reorganised static templates and logic

- Added 404 and 500 error handlers using factory pattern
- Added 404 and 500 error page HTML templates
- Renders survey long name for context
- Fixed some broken logic with ajax due to previous refactors
- Remove nocache endpoint
